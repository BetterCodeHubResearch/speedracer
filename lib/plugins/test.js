const path = require('path')

const { createDir } = require('../.internal/util')

export default (options) => (reports, file) => {
  reports.forEach(report => {
    const dirname = path.join(path.dirname(file), options.snapshotsDirname)
    const filename = path.join(dirname, `${report.metadata.id}.speedracer`)

    return loadReport(filename).then(
      oldReport => {
        process.exit(0)
      },
      () => createDir(dirname).then(
        () => writeFile(filename, race.report.serialize())
      )
    )
  })
}
