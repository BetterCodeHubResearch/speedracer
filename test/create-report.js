// Native
import zlib from 'zlib'

// Packages
import test from 'ava'
import pify from 'pify'

// Ours
import createReport from '../lib/create-report'
const { readFile } = require('../lib/.internal/util')

const loadTrace = filename =>
  readFile(filename, null)
    .then(pify(zlib.gunzip))
    .then(JSON.parse)

test('report profiling', async t => {
  const trace = await loadTrace('./test/fixtures/high-cpu.trace.gz')
  const report = await createReport('./test/fixtures/high-cpu.trace', trace)
  t.snapshot(report.profiling)
})

test('report fps', async t => {
  const trace = await loadTrace('./test/fixtures/render-loop.trace.gz')
  const report = await createReport('./test/fixtures/render-loop.trace', trace)
  t.snapshot(report.fps)
})

test('report first paint', async t => {
  const trace = await loadTrace('./test/fixtures/render-loop.trace.gz')
  const report = await createReport('./test/fixtures/render-loop.trace', trace)
  t.snapshot(report.firstPaint)
})
