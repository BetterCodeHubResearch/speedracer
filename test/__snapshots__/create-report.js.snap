// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`report first paint 1`] = `0.020369`;

exports[`report fps 1`] = `
Object {
  "hi": 68.34,
  "lo": 54.01,
  "mean": 60.21,
  "sd": 1.69,
  "variance": 2.87,
}
`;

exports[`report profiling 1`] = `
Object {
  "categories": Object {
    "loading": 4.254999995231628,
    "painting": 0.2640000134706497,
    "rendering": 0.7060000002384186,
    "scripting": 2244.7900000065565,
  },
  "events": Object {
    "Compile Script": 0.6249999850988388,
    "Composite Layers": 0.21800000965595245,
    "Evaluate Script": 3.7090000212192535,
    "JS Frame": 2240.4560000002384,
    "Layout": 0.29500000178813934,
    "Paint": 0.046000003814697266,
    "Parse HTML": 4.254999995231628,
    "Recalculate Style": 0.18800000846385956,
    "Update Layer Tree": 0.22299998998641968,
  },
  "functions": Object {
    "CompositeLayers": 0.21800000965595245,
    "EvaluateScript": 3.7090000212192535,
    "Layout": 0.29500000178813934,
    "Paint": 0.046000003814697266,
    "ParseHTML": 4.254999995231628,
    "UpdateLayerTree": 0.22299998998641968,
    "UpdateLayoutTree": 0.18800000846385956,
    "f:@24": 1.3379999995231628,
    "f:@25": 0.17599999904632568,
    "f:isPrime@25": 55.21499998867512,
    "f:searchPrimes@25": 2183.727000012994,
    "v8.compile": 0.6249999850988388,
  },
}
`;
